crates/rust-analyzer/src/handlers.rs

```diff
@@ -844,6 +844,21 @@ pub(crate) fn handle_formatting(
 
     let mut rustfmt = match &snap.config.rustfmt {
         RustfmtConfig::Rustfmt { extra_args } => {
+            let mut cmd = process::Command::new(toolchain::rustfmt());
+            cmd.arg("--help");
+            let try_cmd_result = cmd.stdin(Stdio::null()).stdout(Stdio::null()).output()?;
+            if !try_cmd_result.status.success() {
+                let mut rustup = process::Command::new(toolchain::rustup());
+                rustup.args(&["component", "add", "rustfmt"]);
+                let out = rustup.output()?;
+                if !out.status.success() {
+                    snap.user_error("Cannot run rustfmt, cannot install it either".into());
```

Comment:
Can you show the actual install error?

Reply:
Yes, I could capture that and include it in the message.  Do you know if multi-line messages render as plain text, or is it markdown or what?

Reply:
I don't know. To test it you could use a toolchain for which you can't install rustfmt like a locally build rust install, an install using rustup-toolchain-install-master or a nightly version that has rustfmt missing.

Reply:
I've put in the stderr and stdout from calling rustup if it fails (and tested that it comes through using a system toolchain without rustfmt) but sadly vscode's notifications don't support newlines at this time, so it's a bit run-together - at least the info is there though.